core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -216,8 +208,15 @@ public void partitionPageByColumn(Page page)
                 partitionPositions.clear();
             }
         }
+    }
 
-        flushPositionsAppenders(false);
+    private long getPositionsAppendersSizeInBytes()
```

Comment:
Could it be expensive potentially?

Reply:
not expensive but it can cost something in case of a large number of columns/partitions/big row type column